src/mn_menu.c

```diff
@@ -2608,6 +2608,18 @@ boolean M_ShortcutResponder(const event_t *ev)
         }
     }
 
+    if (M_InputActivated(input_menu_prevlevel))
+    {
+        if (demoplayback && singledemo && !PLAYBACK_SKIP)
```

Comment:
Not sure about this part.

Reply:
"Next Level" works during `demoplayback`. "Prev Level" would not be so practical - we would have to restart the demo from the beginning. I think we need demo compatible save/rewind to implement the "Prev Level" feature well.

Reply:
> I think we need demo compatible save/rewind to implement the "Prev Level" feature well.

For demos, yes. But for regular play?

What if you start the game with `-warp 1 2` and hit the "prev level" button? I would expect it to go back to E1M1, although this map has not been visited in the current run.

Reply:
> For demos, yes. But for regular play?

Yes, I was talking about demos. Sure, we can implement "prev map" without demo playback support.

Reply:
Demo support for "prev level" would be very useful.

Reply:
> Demo support for "prev level" would be very useful.

Then it will take a while until we have demo-compatible saves. DSDA-Doom has an implementation of it, but I want to try to extend zone memory with "relative pointers" idea: https://nullprogram.com/blog/2023/09/30/
> instead of pointers I use signed integers whose value indicates an offset relative to itself

This would allow us to save/restore the entire playsim with a single `memcpy`. Not sure if this will work though.